### PR TITLE
[msbuild] Link storyboards as part of the IBTool task if Xcode >= 7.0

### PR DESCRIPTION
**Pull request by jstedfast**
Fixes issue #40583

### CODE REVIEW

File: msbuild/tests/Xamarin.iOS.Tasks.Tests/TargetTests/TargetTests.cs [hunk 1]
```diff
@@ -435,10 +435,14 @@ public void UnpackLibraryResources_LibraryProject ()
 		public void BundleResources ()
 		{
 			var actool = Path.Combine ("obj", "iPhoneSimulator", "Debug", "actool", "bundle");
-			var ibtool = Path.Combine ("obj", "iPhoneSimulator", "Debug", "ibtool");
-
```
Comment (mandel-macaque) [inline msbuild/tests/Xamarin.iOS.Tasks.Tests/TargetTests/TargetTests.cs]: Not a big issue, but I think you can do:

```csharp
var ibtool = Path.Combine ("obj", "iPhoneSimulator", "Debug", AppleSdkSettings.XcodeVersion.Major >= 7 ? "ibtool-link": "ibtool");
```

File: msbuild/Xamarin.MacDev.Tasks.Core/Tasks/IBToolTaskBase.cs [hunk 2]
```diff
@@ -218,21 +221,24 @@ public override bool Execute ()
...
-					outputDir = Path.GetDirectoryName (rpath);
-					output = new TaskItem (rpath);
+				output.SetMetadata ("LogicalName", bundleName);
+				output.SetMetadata ("Optimize", "false");
 
-					output.SetMetadata ("LogicalName", bundleName);
+				if (Path.GetExtension (bundleName) != ".plist") {
+					// Don't include Watch storyboards that got compiled to plists
+					compiled.Add (output);
 				}
```
Comment (mandel-macaque) [inline msbuild/Xamarin.MacDev.Tasks.Core/Tasks/IBToolTaskBase.cs]: Style: Extra {} (I think, I'm not great with the mono style, if not, sorry for the noise).

**Comment by rolfbjarne** [comment]
This breaks watchos app projects, the storyboard isn't copied to the final app.

Build logs (both working and failing): https://gist.github.com/2a528e3cb8239980b57aed691c6f2634

Working build results in this:

    $ find . | grep Interface
    ./bin/iPhoneSimulator/Debug-watchos-app/App.app/Interface.plist
    ./obj/iPhoneSimulator/Debug-watchos-app/ibtool/Interface.storyboardc/Interface.plist
    ./obj/iPhoneSimulator/Debug-watchos-app/ibtool-link/Interface.plist
    ./obj/iPhoneSimulator/Debug-watchos-app/ibtool-manifests/Interface.plist

Broken build:

    $ find . | grep Interface
    ./obj/iPhoneSimulator/Debug-watchos-app/ibtool/Interface.storyboardc/Interface.plist
    ./obj/iPhoneSimulator/Debug-watchos-app/ibtool-manifests/Interface.plist

**Comment by jstedfast** [comment]
Fixed in https://github.com/xamarin/xamarin-macios/pull/75

**Pull request merged**